### PR TITLE
ctlv2: allow to add a user within one command line

### LINKED ISSUE #6668: Cannot create a new user within just one command input
Currently, the etcdctl that are using the ctlv2, at least, are not allowing you to create a user specifying the password within the same command line.

Doing etcdtcl user add root:toor

Will result in etcdctl waiting for you to ingress a new password in a new prompt line.

root@jbazan-virtual-machine:/workspace# etcdctl user add root:toor
New password:

This makes a little bit more difficult automating infrastructure creation, like cloud config scripts, or cloudformation, etc...

I have a branch that makes this functional for ctlv2-based ctl (the one that I'm using).
If you think this is a valid issue, I'll go right away and create the pull request.

Cheers,
Jonathan

### PR DESCRIPTION
**Pull request by johnbazan**
This makes the "user add usr:pwd" feature available for ctlv2
without asking for the password in a new prompt.

Fixes #6668

### CODE REVIEW

**Comment by gyuho** [comment]
@johnbazan  We do not plan on releasing any more `v2.3.X` unless it needs bug fix release.

Any thought? @mitake @xiang90

**Comment by xiang90** [comment]
> I was just looking forward to make a contribution, not actually looking for a new release, hehe.

@gyuho suggests that you might need to build from etcd master or to use any release after today to include this feature even if we get it merged.

If you do not care about when it can get in, we do not need to worry either.

**Comment by gyuho** [comment]
@johnbazan Right. I am not against this PR :)
Just wanted to note that this feature is not likely to be in any v2 release.

**Comment by johnbazan** [comment]
I absolutely can live with that. :)  hehe

File: etcdctl/ctlv2/command/user_commands.go [hunk 1]
```diff
@@ -111,18 +111,20 @@ func actionUserAdd(c *cli.Context) error {
 	api, userarg := mustUserAPIAndName(c)
 	ctx, cancel := contextWithTotalTimeout(c)
 	defer cancel()
-	user, _, _ := getUsernamePassword("", userarg+":")
+
+	user, pass, err := getUsernamePassword("New password: ", userarg)
+
```
Comment (gyuho) [inline etcdctl/ctlv2/command/user_commands.go]: remove this blank line?

File: etcdctl/ctlv2/command/user_commands.go [hunk 2]
```diff
@@ -111,18 +111,20 @@ func actionUserAdd(c *cli.Context) error {
...
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error reading user/password:", err)
+		os.Exit(1)
+	}
 	currentUser, err := api.GetUser(ctx, user)
+
```
Comment (gyuho) [inline etcdctl/ctlv2/command/user_commands.go]: remove this blank line?

**Comment by mitake** [comment]
@johnbazan @gyuho this change seems to be reasonable to me. BTW, the commit title should be `etcdctl: allow to add a user within one command line`?

**Comment by johnbazan** [comment]
Sure. I'll change it, add the tests and let you know.

**Comment by xiang90** [comment]
Closing. this does not seem like a problem anymore.

**Comment by johnbazan** [comment]
Sorry guys, I screwed up.
Mixed some versions and this patch doesn't really fix anything.
It just reorganizes code.
The tests already include one with this flow and works fine with the latest master.

**Comment by johnbazan** [comment]
I've found another one,

you can actually create an empty user today.

etcdctl user add :password

**Comment by xiang90** [comment]
@johnbazan Probably send another pr to fix it.

**Comment by johnbazan** [comment]
Just before I create a new issue with something wrong...

etcdctl is alowing me to create an emtpy user.

etcdctl user add :pass 

Will create an empty named user.
Then, you wont be able to delete it,
since etcdctl user remove ""     

returns 

jbazan@jbazan-virtual-machine:~/etcd/bin$ ./etcdctl user remove ""
invalid character 'M' looking for beginning of value

**Pull request closed without merging**